stan/math/prim/scal/err/is_size_match.hpp

```diff
@@ -0,0 +1,30 @@
+#ifndef STAN_MATH_PRIM_SCAL_ERR_IS_SIZE_MATCH_HPP
+#define STAN_MATH_PRIM_SCAL_ERR_IS_SIZE_MATCH_HPP
+
+#include <boost/type_traits/common_type.hpp>
+#include <stan/math/prim/scal/meta/likely.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Check if the provided sizes match
```

Comment:
Similar comment as above.

Reply:
I think this one should be documented a little differently because we're not actually passing in variables.

How's something like this?
```
Returns <code>true</code> if the provided sizes match.

@tparam T_size1 Type of the first provided size. Expecting this to be <code>int</code> or <code>size_t</code>.
@tparam T_size2 Type of the second provided size. Expecting this to be <code>int</code> or <code>size_t</code>.
```
And leaving the rest pretty much alone.